packages/optimizer/lib/transformers/AutoExtensionImporter.js

```diff
@@ -0,0 +1,269 @@
+        existingImports.add(customElement);
+      }
+      // Explicitly detect amp-access via the script tag in the header
+      if (node.tagName === 'script' && node.attribs['id'] === 'amp-access') {
+        extensionsToImport.add('amp-access');
+      }
+      node = node.nextSibling;
+    }
+  }
+
+  async findExtensionsToImportInBody_(body, extensionsToImport) {
+    let node = body;
+    const promises = [];
+    while (node !== null) {
+      if (node.tagName) {
+        promises.push(this.addRequiredExtensionByTag_(extensionsToImport, node));
+        promises.push(this.addRequiredExtensionByAttributes_(extensionsToImport, node));
+      }
+      node = node.nextNode();
+    }
+    return Promise.all(promises);
+  }
+
+  async addRequiredExtensionByTag_(allRequiredExtensions, node) {
+    const extensionSpec = (await this.extensionSpec);
+    // Check for required extensions by tag name
+    const requiredExtensions = extensionSpec.tagToExtensionsMapping.get(node.tagName);
+    if (requiredExtensions) {
+      requiredExtensions.forEach((ext) => allRequiredExtensions.add(ext));
+    }
+    // Add custom templates (e.g. amp-mustache)
+    if (node.tagName === 'template' && node.attribs.type) {
+      allRequiredExtensions.add(node.attribs.type);
+    }
+  }
+
+  async addRequiredExtensionByAttributes_(allRequiredExtensions, node) {
+    if (!node.tagName || !node.attribs) {
+      return;
+    }
+    // Look for element attributes indicating AMP components (e.g. amp-fx)
+    const extensionSpec = await this.extensionSpec;
+    const tagToAttributeMapping = extensionSpec.tagToAttributeMapping;
+    const attributesForTag = tagToAttributeMapping.get(node.tagName);
+    if (attributesForTag) {
+      attributesForTag.forEach((attribute) => {
+        if (node.attribs[attribute.name] !== undefined) {
+          attribute.requiresExtension.forEach((ext) => {
+            allRequiredExtensions.add(ext);
+          });
+        }
+      });
+    }
+    // Check for amp-bind attribute bindings
+    const tagToBindAttributeMapping = extensionSpec.tagToBindAttributeMapping;
+    const attributeNames = Object.keys(node.attribs);
+    if (attributeNames.some((a) => a.startsWith('[') || a.startsWith('data-amp-bind'))) {
+      allRequiredExtensions.add('amp-bind');
+    }
+    // Rewrite short-form `bindtext` to `data-amp-bind-text`
+    // to avoid false-positives we check for each tag only the
+    // supported bindable attributes (e.g. for a div only bindtext, but not bindvalue).
+    const ampBindAttrs = tagToBindAttributeMapping.get(node.tagName);
+    // true if we need to import amp-bind
+    let usesAmpBind = false;
+    for (const attributeName of attributeNames) {
+      if (!attributeName.startsWith(BIND_SHORT_FORM_PREFIX)) {
+        continue;
+      }
+      const attributeNameWithoutBindPrefix =
+        attributeName.substring(BIND_SHORT_FORM_PREFIX.length);
+
+      // Rename attribute from bindx to data-amp-bind-x
```

Comment:
Out of curiosity, why?

Reply:
VS Code does not autocomplete `data-X` attributes. With `bindX` you get component specific autocompletion for bindable attributes.